seacrowd/sea_datasets/copal/copal.py

```diff
@@ -0,0 +1,154 @@
+providing a more natural portrayal of day-to-day causal reasoning within the Indonesian cultural sphere.
+Professionally written by natives from scratch, COPAL-ID is more fluent and free from awkward phrases, unlike the translated XCOPA-ID.
+Additionally, COPAL-ID is presented in both standard Indonesian and Jakartan Indonesian–a commonly used dialect.
+It consists of premise, choice1, choice2, question, and label, similar to the COPA dataset.
+"""
+
+_HOMEPAGE = "https://huggingface.co/datasets/haryoaw/COPAL"
+
+_LICENSE = Licenses.CC_BY_SA_4_0.value
+
+_URLS = {
+    _DATASETNAME: {"test": "https://huggingface.co/datasets/haryoaw/COPAL/resolve/main/test_copal.csv?download=true", "test_colloquial": "https://huggingface.co/datasets/haryoaw/COPAL/resolve/main/test_copal_colloquial.csv?download=true"},
+}
+
+_SUPPORTED_TASKS = [Tasks.COMMONSENSE_REASONING]
+
+_LOCAL = False
+_LANGUAGES = ["ind"]
+
+_SOURCE_VERSION = "1.0.0"
+
+_SEACROWD_VERSION = "1.0.0"
+
+
+class copal(datasets.GeneratorBasedBuilder):
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="copal_source",
+            version=SOURCE_VERSION,
+            description="COPAL test source schema",
+            schema="source",
+            subset_id="copal",
+        ),
+        SEACrowdConfig(
+            name="copal_colloquial_source",
+            version=SOURCE_VERSION,
+            description="COPAL test colloquial source schema",
+            schema="source",
+            subset_id="copal",
+        ),
+        SEACrowdConfig(
+            name="copal_seacrowd_qa",
+            version=SEACROWD_VERSION,
+            description="COPAL test seacrowd schema",
+            schema="seacrowd_qa",
+            subset_id="copal",
+        ),
+        SEACrowdConfig(
+            name="copal_colloquial_seacrowd_qa",
+            version=SEACROWD_VERSION,
+            description="COPAL test colloquial seacrowd schema",
+            schema="seacrowd_qa",
+            subset_id="copal",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "copal_source"
+
+    def _info(self):
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "choice1": datasets.Value("string"),
+                    "choice2": datasets.Value("string"),
+                    "question": datasets.Value("string"),
+                    "idx": datasets.Value("int64"),
+                    "label": datasets.Value("int64"),
+                    "terminology": datasets.Value("int64"),
+                    "culture": datasets.Value("int64"),
+                    "language": datasets.Value("int64"),
+                }
+            )
+        elif self.config.schema == "seacrowd_qa":
+            features = schemas.qa_features
```

Comment:
since `COPAL` has columns like `terminology`, `culture`, and `language` which can be useful for evaluation, do you mind to add those into `features["meta"]` field?

it can be added as follows:
```python
features["meta"] = {
   'terminology': datasets.Value("int64"),
   'culture': datasets.Value("int64"),
   'language': datasets.Value("int64")}
```

and add the data aswell in `_generate_examples()` like:
```python
"meta": {
    "terminology": row.Terminology,
    "culture": row.Culture,
    "language": row.Language
}
```

Reply:
done, thanks for the snippets 👍